### DISCUSSION TITLE
[Design/Antagonist] Vampire Rework

### DISCUSSION THREAD

**Discussion by dearmochi** [ParadiseSS13/Paradise#16268] [Documents]
# Vampire Rework

### Motivation

Vampire gameplay (as and against) is boring and tedious. It's either get caught at the first victim or be unstoppable with constant misting. It's not fun and challenging.

Most glaring issues:
- Too easy to reach full power
  - Blood bag abuse
- Imbalance of abilities
  - Hypnotize, Shapeshift, Diseased Touch are basically useless
  - Glare and Screech are 1-click-win stuns
  - Shadowstep and Mist allow for too much mobility
- Objectives do not match the quick power creep
  - Especially with steal objectives, misting makes it ridiculously easy
- Linear ability progression, gets boring quickly
- Early game is tough but snowballing happens very easily with some abilities
- Lack of counterplay

### Desired Outcome

A complete rework of the Vampire antagonist that aims at giving them a longer curve towards full power while providing more variety to their abilities.

## Design

<details>
<summary>Mechanics</summary>

### Mechanic: Blood Pool

A vampire's resource, gained by drinking blood of non-mindless humans and spent by using abilities or unlocking them.

To drink blood, the vampire has to at least passively grab the target, then attack them on Harm intent while targeting the head. Doing so will:
- Initiate a progress bar which, upon completion will add to the vampire's blood pool while removing some blood from the victim.
  - The progress bar will interrupt for any of the generic reasons (incapacitation, distance, movement) or if the victim is completely out of blood.
- Play a somewhat quiet biting sound.
- Send a visible message indicating of the action.

The amount of resource gained depends on whether the victim is alive or not.

### Mechanic: Blood Thirst

Instead of subsisting on food, vampires need a regular intake of blood, be it via biting a victim or ingesting blood, the latter of which is inefficient as it will not increase their blood pool.

### Mechanic: Specialization

The first choice a vampire will make upon acquiring sufficient blood, around 100.

Specialization offers the vampire a variety of paths, each with a different set of (unlockable) abilities that are suited for a playstyle. The available abilities become more powerful as the vampire progresses down the ability tree.

Picking a specialization is optional but highly recommended. Doing so will lock the vampire into it, and they may not undo their choice.
</details>

<details>
<summary>Passives</summary>

### Passive: Holy Weakness

*Innate*

Vampires are naturally weak to holy things, and exposure can range from inconvenient to fatal.

- A vampire entering the Chapel will be on hallowed ground, constantly dealing burn damage. After a few cycles, they will be set on fire.
  - There should be a visual alert when standing on hallowed ground.
- The Chaplain is a vampire's archenemy:
  - Glaring the Chaplain will be reflected back to the vampire, stunning themselves.
  - Most vampiric abilities otherwise do not affect the Chaplain.
  - Drinking the Chaplain's blood by biting their neck will deal considerable burn damage in addition of subtracting some blood from the vampire's blood pool.
- Holy Water behaves differently, depending on method of application:
  - If ingested, Holy Water deals burn damage to the vampire and depletes their blood pool at a constant rate. If their blood pool is empty, the vampire catches fire.
  - If splashed on the vampire, they will suffer burn damage proportional to the amount splashed.
- The Bible can be used as a weapon against the vampire, dealing bonus burn damage. If picked up by the vampire, they will catch fire and be incapacitated for a short while.
- Being in space, vampires are exposed to sunlight (or starlight). Even with proper equipment, a vampire bathing in starlight will receive the same effects as they would while standing on hallowed ground.

### Passive: Necropolis Interference

*Innate*

While in Lavaland, a vampire's blood sucking ability is crippled. 

While they are able to drink blood while on the planet, only a small amount can increase their blood pool, just enough to select a specialization.  Drinking any more blood will only sate the vampire's thirst, but nothing more.

### Passive: Vampiric Vision

*300 Blood*

Grants the vampire thermal vision.

### Passive: Full Power

*1,000 Blood*

Grants the vampire immunity to the Chapel's effects and as well as the ability to bypass the Chaplain's holy protection offensively.
</details>

<details>
<summary>Abilities</summary>
### Ability: Rejuvenate

*Innate, One-off*

Rejuvenate instantly removes any immobilizing effects from the vampire as well as a large portion of stamina damage.

### Ability: Glare

*Innate, One-off*

Cost: 0
Cooldown: Medium

Casting Glare will:
- Weaken any living human facing the vampire for a moderate duration. The duration is reduced if the victim isn't fully facing the vampire.
  - A full glare will, in addition, mute the victim for a brief period.
- Briefly show a red, glowing glint on the vampire's eyes, ignoring clothing.
- Display a visible message indicating of the action, in addition to a danger message to any victims.

### Ability: Hypnotize

*Innate, Channeled*

Cost: 0
Cooldown: Long
Cast time: Short

At the end of the duration, puts the target to sleep for a moderate amount of time.

### Ability: Shadowstep

*300 Blood, One-off*

Cost: 30
Cooldown: Short

Teleports the vampire to a random dark tile in range. Will not function if no applicable tile is found.
</details>

## Specialization: Hemomancer

Generalist specialization, based on blood manipulation.

<details>
<summary>Abilities</summary>

### Ability: Crimson Claws

*300 Blood, One-off*

Cost: 50
Cooldown: Medium

Conjures deadly, sharp claws out of blood which convert a portion of damage dealt into usable blood, if the victim is compatible (non-mindless human). The claws has unlimited uses until dropped, after which it will disappear.

### Ability: Blood Spike

*600 Blood, One-off, Single Target*

Cost: 25
Cooldown: Short

Upon targeting a visible turf, a spike made of blood erupts after a short time, damaging anyone on the turf.

### Ability: Pool Form

*600 Blood, One-off*

Cost: 75
Cooldown: Long

Transforms the vampire into a pool of blood for a short time, allowing them to move freely through flooring, airlocks and fake walls and leaving blood puddles behind them.

### Ability: Eruption

*1,000 Blood, One-off*

Cost: 100
Cooldown: Long

Blood spikes emerge from any blood pool within the vampire's field of view,  dealing moderate damage to anyone standing on them, as well as stunning them.
</details>

## Specialization: Gargantua

Focuses on taking and dishing damage.

<details>
<summary>Abilities</summary>

### Ability: Blood Swell

*300 Blood, Toggle*

Cost (initial and every Life/2 seconds): 10
Cooldown: N/A

Slightly enlarges the vampire, granting them a slight reduction to stuns and stamina damage while active.

### Ability: Overwhelming Force

*600 Blood, Toggle*

Cost (upon successful activation): 50
Cooldown: Short

While toggled, bumping into an unbolted airlock will automatically pry it open, consume usable blood and toggle the ability off again.

### Ability: Sanguine Grasp

*1,000 Blood, One-off, Single Target*

Cost: 75
Cooldown: Moderate

Prepares a neck grab on the vampire's active hand. Upon hitting a human with it, they will be instantly neck grabbed. While grabbing someone this way, the vampire will drain blood at a quick rate while also dealing brute damage to the victim.

### Ability: Dark Mending

*1,000 Blood, Channeled*

Cost: 100
Cooldown: Moderate
Cast time: Moderate

Upon completion, rejuvenates the vampire's own targeted limb.
</details>

<details>
<summary>Upgrades</summary>

### Upgrade: Blood Swell II

*600 Blood*

Grants moderate stun and stamina damage reduction, in addition to a slight brute and burn damage reduction. Confers a flat brute damage bonus to any attack, armed or not by 5. Increases cost to 15.

### Upgrade: Blood Swell III

*1,000 Blood*

Grants high stun and stamina damage reduction, in addition to a moderate brute and burn damage reduction. Brute bonus increased to 10. Increases cost to 20.
</details>

## Specialization: Dantalion

Revolves around making thralls, akin to mindslaves.

<details>
<summary>Abilities</summary>

### Ability: Enthrall

*300 Blood, Channeled, Single Target*

Cost: 150
Cooldown: High
Cast time: Moderate

Upon completion, the targeted victim becomes a thrall to the vampire. Thralls may not be used as sources of blood for vampires, and they incur additional blood thirst on their master, as upkeep. Thralls have to obey their master's every command until deconverted by being fed Holy Water for some time.
Upon examination, thralls have a tell indicating that they are thralled.

Cannot be cast on thralls, mindless or mindshielded people.

### Ability: Blood Link

*300 Blood, One-off*

Cost: 0
Cooldown: None

Allows the vampire to telepathically communicate with their thralls, regardless of distance.

### Ability: Mesmerize

*600 Blood, One-off, Single Target*

Cost: 50
Cooldown: Medium

Causes the target to be temporarily hypnotized, walking towards the vampire for the duration of the effect. Their character may not be controlled during that time.

### Ability: Pacify

*600 Blood, One-off, Single Target*

Cost: 50
Cooldown: Medium

Causes the target to temporarily pacified, becoming unable to attack anyone for the duration.

### Ability: Subjugation

*1,000 Blood, One-off*

Cost: 100
Cooldown: Long

Weakens all visible, non-thrall humans for a moderate period.

### Ability: Rally Thralls

*1,000 Blood, One-off*

Cost: 150
Cooldown: Long

Rejuvenates all visible thralls like the vampire Rejuvenate ability.
</details>

## Specialization: Umbrae

Revolves around darkness, providing mobility and advantages.

<details>
<summary>Passives</summary>

### Passive: Cloak of Darkness II

Provides a slight chance of letting projectiles aimed at the vampire pass through.
</details>

<details>
<summary>Abilities</summary>

### Ability: Cloak of Darkness

*300 Blood, Toggle*

Cost: 0
Cooldown: None

While toggled, being in darkness provides the vampire significant invisibility.

### Ability: Shadow Snare

*300 Blood, One-off, Turf*

Cost: 50
Cooldown: Medium

Places a partially invisible trap at a given location. When triggered by a non-vampire human, they will become snared and blinded for a moderate duration.

The trap may only be placed in a dark tile. Upon exposure to light, it will be revealed and disappear after enough exposure. The trap lasts for about 5 minutes.

### Ability: Extinguish

*600 Blood, One-off*

Cost: 50
Cooldown: Medium

Extinguishes all nearby light sources, providing darkness for the vampire. Glowing plants are destroyed.

### Ability: Eternal Darkness

*1,000 Blood, Toggle*

Cost: 0
Cooldown: None

While toggled, covers the vampire's surroundings in a shroud of darkness, blinding and slowing any non-vampire human remaining inside.
</details>

## Objectives

Upon picking a specialization, the vampire should receive a set of appropriate objectives given their abilities.

## Ability Tree

<table>
  <tr style="font-weight: bold">
    <td>Unlock</td>
    <td style="background-color: #e74c3c; color: white">Hemomancer</td>
    <td style="background-color: #e67e22; color: white">Gargantua</td>
    <td style="background-color: #8e44ad; color: white">Dantalion</td>
    <td style="background-color: #2c3e50; color: white">Umbrae</td>
  </tr>
  <tr>
    <td><b>0</b></td>
    <td colspan="4" align="center">Glare<br />Hypnotize<br />Rejuvenate</td>
  </tr>
  <tr>
    <td><b>100</b></td>
    <td colspan="4" align="center"><b>Specialization Choice</b></td>
  </tr>
  <tr>
    <td>300</td>
    <td>Crimson Claws</td>
    <td>Blood Swell</td>
    <td>Enthrall<br />Blood Link</td>
    <td>Cloak of Darkness<br />Shadow Snare</td>
  </tr>
  <tr>
    <td><b>300</b></td>
    <td colspan="4" align="center"><b>Vampiric Vision</b><br />Shadowstep</td>
  </tr>
  <tr>
    <td>600</td>
    <td>Blood Spike<br />Pool Form</td>
    <td>Blood Swell II<br />Overwhelming Force</td>
    <td>Mesmerize<br />Pacify</td>
    <td>Dark Mist<br />Extinguish</td>
  </tr>
  <tr>
    <td>1,000</td>
    <td>Eruption</td>
    <td>Blood Swell III<br />Sanguine Grasp<br />Dark Mending</td>
    <td>Subjugation<br />Rally Thralls</td>
    <td>Cloak of Darkness II<br />Eternal Darkness</td>
  </tr>
  <tr>
    <td><b>1,000</b></td>
    <td colspan="4" align="center"><b>Full Power</b></td>
  </tr>
</table>

## Non-Code Work

- Writing, giving each spell name and description a theme
- Sprite work -- spell icons, animations

**Comment by SteelSlayer** [comment]
I left a couple comments on the HackMD but I'll leave them here as well.

- For Umbrae's Extinguish ability: the 50 blood cost seems a very high for what it does.
- For the Hemomancer's Blood Spike ability: I think that having to target a turf and the fact that the ability has small delay will make it hard to use on most people in combat, unless they're standing still. Or perhaps this is mainly intended to be combo'd with a stun like glare?

**Comment by Necaladun** [comment]
This is fucking great all round and highly encourage it. They need a rework for sure.

I'd really like to see a move away from teleporting - it's by far the most irritating and powerful ability. The hemomancer having some but limited I like, possibly even akin to blood crawl, so there is a way to counter it more easily?

The main other 'class' suggestion I can think of is a gangrel-esque one, with abilities around bats/wolves/shapeshifting etc.

**Comment by ghost** [comment]
This is cool someone should code it 😏

**Reply by hal9000PR** [reply]
done :)

**Comment by SabreML** [comment]
Added in #16476.
